header.go

```diff
@@ -43,6 +43,61 @@ type Header struct {
 	rawTLVs            []byte
 }
 
+// HeaderProxyFromAddrs creates a new PROXY header from a source and a
+// destination address. The latest protocol version is used.
+func HeaderProxyFromAddrs(sourceAddr, destAddr net.Addr) *Header {
+	h := &Header{
+		Version: 2,
+		Command: PROXY,
+	}
+	switch sourceAddr := sourceAddr.(type) {
+	case *net.TCPAddr:
+		destAddr, ok := destAddr.(*net.TCPAddr)
+		if !ok {
+			break
```

Comment:
Should we return an error instead?

Reply:
I've chosen to never return an error and fill the fields in a best-effort fashion. In this case, that means setting `TransportProtocol` to `UNSPEC` if the function can't figure out what to do.

I wonder if it makes sense to return an error instead. What would the caller do with the error? Making the error fatal because we can't infer the right details wouldn't be great.

Reply:
Well, I think this reasoning should be part of the function comment so consumer knows exactly what the contract is: a best-effort helper.

Reply:
Good point, will expand the docs.